src/index.ts

```diff
@@ -39,22 +39,69 @@ function logError(msg) {
   spinner.fail();
 }
 
-function transformWebModuleFilename(depName: string): string {
-  return depName.replace('/', '--');
+class ErrorWithHint extends Error {
+  constructor(message: string, public readonly hint: string) {
+    super(message);
+  }
+}
+
+/**
+ * Resolve a "webDependencies" input value to the correct absolute file location.
+ * Supports both npm package names, and file paths relative to the node_modules directory.
+ * Follows logic similar to Node's resolution logic, but using a package.json's ESM "module"
+ * field instead of the CJS "main" field.
+ */
+function resolveWebDependency(dep: string): string {
+  const lazyDependencyPath = path.join(cwd, 'node_modules', dep);
+  let dependencyStats: fs.Stats;
+  try {
+    dependencyStats = fs.statSync(lazyDependencyPath);
+  } catch (err) {
+    throw new Error(`"${dep}" not found in your node_modules directory. Did you run npm install?`);
+  }
+  if (dependencyStats.isFile()) {
+    return lazyDependencyPath;
+  }
+  if (dependencyStats.isDirectory()) {
+    const dependencyManifestLoc = path.join(cwd, 'node_modules', dep, 'package.json');
+    const manifest = require(dependencyManifestLoc);
+    if (!manifest.module) {
+      throw new ErrorWithHint(
+        `dependency "${dep}" has no ES "module" entrypoint.`,
+        chalk.italic(
+          `Tip: Find modern, web-ready packages at ${chalk.underline(
+            'https://pikapkg.com/packages',
+          )}`,
+        ),
+      );
+    }
+    return path.join(cwd, 'node_modules', dep, manifest.module);
+  }
+
+  throw new Error(
+    `Error loading "${dep}" at "${lazyDependencyPath}". (MODE=${dependencyStats.mode}) `,
+  );
+}
+
+/**
+ * Formats the @pika/web dependency name from a "webDependencies" input value:
+ * 1. Replace all `/` URL/path seperators with "--"
+ * 2. Remove any ".js" extension (will be added automatically by Rollup)
```

Comment:
'.js' may not be an artefact of a file extension; it may actually be part of the package's name (see popper.js). in which case I might intuit that `import 'popper.js'` should become 'import '/web_modules/popper.js.js''.

this is why I was careful to give different handling to modules identified via package name vs modules explicitly pointed to via paths.

Reply:
I think `popper.js.js` would come off as strange. Agreed that we could maybe be more careful about what happens if you're installing both "popper" and "popper.js", but my guess is these cases are very rare.

Reply:
I agree that we may _get lucky_ as far as `.js` clashes are concerned. just like we may _get lucky_ with `/` to `--` clashes.

But I think this is only superficially helpful to the user. I'm not certain what their intuition would be. A rule like "always add one more .js" is more straightforward than "add a .js unless you get one for free".